### PR TITLE
Sanitize `None` params during pruning

### LINKED ISSUE #6835: `None` parameters not sanitized during pruning
## 🐛 Bug

ModelPruning callback fails when a module parameter is `None`. This can happen, for instance,  in a `Linear()` when `bias=False`. 

## Please reproduce using the BoringModel
https://colab.research.google.com/drive/1UApprg-5htIQbosiSyyLLXm1B8wE8EbN?usp=sharing

### Expected behavior

Unavailable parameters are already taken care of in `sanitize_parameters_to_prune` method in `ModelPruning` callback. A trivial check to ensure the attribute, when present, is not `None` should solve this. Happy to send a PR.

### Environment
* CUDA:
	- GPU:
		- Tesla K80
	- available:         True
	- version:           10.1
* Packages:
	- numpy:             1.19.5
	- pyTorch_debug:     False
	- pyTorch_version:   1.8.1+cu101
	- pytorch-lightning: 1.2.6
	- tqdm:              4.41.1
* System:
	- OS:                Linux
	- architecture:
		- 64bit
		- 
	- processor:         x86_64
	- python:            3.7.10
	- version:           #1 SMP Thu Jul 23 08:00:38 PDT 2020

### PR DESCRIPTION
**Pull request by karthikprasad**
## What does this PR do?
Fixes #6835 

## Before submitting
- [x] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes? (if necessary)
- [x] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [x] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: pytorch_lightning/callbacks/pruning.py [hunk 1]
```diff
@@ -422,7 +422,10 @@ def sanitize_parameters_to_prune(
         current_modules = [m for m in pl_module.modules() if not isinstance(m, _MODULE_CONTAINERS)]
 
         if parameters_to_prune is None:
-            parameters_to_prune = [(m, p) for p in parameters for m in current_modules if hasattr(m, p)]
+            parameters_to_prune = [
+                (m, p) for p in parameters for m in current_modules
+                if hasattr(m, p) and getattr(m, p) is not None
```
Comment (rohitgr7) [inline pytorch_lightning/callbacks/pruning.py:427]:
```suggestion
                if getattr(m, p, None) is not None
```
will this work?

**Review by justusschock — APPROVED** [review]
Lgtm. Could you also add it to the changelog?

**Review by ananthsub — APPROVED** [review]

**Comment by karthikprasad** [comment]
The two failing tests don't seem to be related to mine? Is there anything I should do to overcome this?

**Review by rohitgr7 — APPROVED** [review]

File: CHANGELOG.md [hunk 2]
```diff
@@ -170,6 +170,9 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
 
 ### Fixed
 
+- Sanitze `None` params during pruning ([#6836])(https://github.com/PyTorchLightning/pytorch-lightning/pull/6836)
```
Comment (rohitgr7) [inline CHANGELOG.md:173]:
```suggestion
- Sanitize `None` params during pruning ([#6836](https://github.com/PyTorchLightning/pytorch-lightning/pull/6836))
```

**Review by carmocca — APPROVED** [review]
> The two failing tests don't seem to be related to mine? Is there anything I should do to overcome this?

They are unrelated. Do not worry about them.

**Pull request merged**